src/lib/marshal/translation/reader.ts

```diff
@@ -62,13 +69,63 @@ const readTranslationFiles = async (
       namespace,
       abspath,
       exists: true,
-      content: JSON.stringify(content),
+      content,
+      format,
     });
   }
 
   return [translations, errors];
 };
 
+const readTranslationFile = async (
+  filePath: string,
+): Promise<
+  [string | undefined, SourceError | undefined, TranslationFormat]
```

Comment:
```suggestion
  [string, undefined, TranslationFormat] | [undefined, SourceError, TranslationFormat]
```
I know just enough TS to get by, so i'm not sure if this will actually work 😅. If too much trouble, then i say don't worry.